pymedphys/_dicom/anonymise/__init__.py

```diff
@@ -133,6 +134,26 @@ def anonymise_dataset(  # pylint: disable = inconsistent-return-statements
         An anonymised version of the input DICOM dataset.
     """
 
+    if replacement_strategy is not None:
+        if "replace_values" in replacement_strategy:
+            replace_values = replacement_strategy["replace_values"]
+        if "keywords_to_leave_unchanged" in replacement_strategy:
+            keywords_to_leave_unchanged = replacement_strategy[
+                "keywords_to_leave_unchanged"
+            ]
+        if "delete_private_tags" in replacement_strategy:
+            delete_private_tags = replacement_strategy["delete_private_tags"]
+        if "delete_unknown_tags" in replacement_strategy:
+            delete_unknown_tags = replacement_strategy["delete_unknown_tags"]
+        if "copy_dataset" in replacement_strategy:
+            copy_dataset = replacement_strategy["copy_dataset"]
+        # unlike the modifiers, if identifying keywords were passed in as parameters, use those
+        if (
+            identifying_keywords is not None
+            and "identifying_keywords" in replacement_strategy
+        ):
+            identifying_keywords = replacement_strategy["identifying_keywords"]
```

Comment:
I think if we're going to go this route, I think we need to raise an error if `replace_values` and the other parameters are defined by the user as well as `replacement_strategy`. To achieve that, I believe we need to default these params to `None` and then raise a `ValueError` if `replacement_strategy` is not None and if any of the parameters that are inherited from `replacement_strategy` are also None.

Reply:
I believe I didn't say the above correctly. But the heart of what I was trying to say is O don't want the user to be able to write:

```python
anonymise(replace_values=False, replacement_strategy=strategy)
```

And then have the code run off and override the parameter that the user passed. I was under the impression we were going to remove those parameters entirely and let replacement_strategy handle them.

Alternatively, if replacement strategy is left as None by the user, and if we want the current API to be able to carry on in unison, then it could be that all those parameters can stay in the function signature, it's just that they're not allowed to be defined by the user if replacement_strategy has also been defined.

Reply:
> I was under the impression we were going to remove those parameters entirely and let replacement_strategy handle them.

Yes.  But I wanted to start by deprecating the current interface, and then follow with obsoleting it.
Further, the appropriate way to get the result of replace_values=False is to create a strategy that provides that approach, and eliminate the replace_values parameter entirely.  Although I can't recommend using such a strategy because it will likely set type 1 parameters to have a value of the empty string, which is not DICOM conformant (and very few applications will succeed with type 1 element non-conformance).

I'm working on putting a facade in front of the previous function calls with the new API (renaming the old ones by prefixing with and underscore to make them private).  That way it's just the CLI that has a directive to clear the values, and that will get the "clear values strategy" and *all* strategies have replace_values=True and any strategy that has a value of False for that will get treated as not valid.  And then I can delete the special case behaviour for that.

I've got some test cases to write.